### PR TITLE
Fix translations input issues

### LINKED ISSUE #9510: Bug when typing in a 2nd-depth-level O2M modal translation field
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

I've got : 
Collection A : 
- field A (O2M) -> Collection B : 
-- field B (O2M) -> Collection C : 
--- field C -> Collection C translations : 
---- field D

In the DIrectus dashboard, from a Collection A item, when I open the sub-sub-modal of Collection C translations, when I start to type in the field D for one of my languages, the focus is systematically blurred.

### To Reproduce

1. From a setup with : 
Collection A : 
- field A (O2M) -> Collection B : 
-- field B (O2M) -> Collection C : 
--- field C -> Collection C translations : 
---- field D

2. Open the 2nd-depth modal to edit field D from the field A.

3. Try to edit the field D

### Errors Shown

https://user-images.githubusercontent.com/17531455/140481333-1e4cfc16-8c48-4ec9-9b08-7bbeacc83c70.mov

### What version of Directus are you using?

v9.0.0 (latest)

### What version of Node.js are you using?

latest

### What database are you using?

MySQL

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying Directus?

Docker

### PR DESCRIPTION
**Pull request by azrikahar**
fixes #9510

## Context

This regression was introduced in #9333, which was aimed at #9210's 3rd sub-issue. There is a follow up fix in #9630, however it seems like it still does not resolve the whole issue at the moment (It doesn't unfocus anytime we type anymore, but the edits get reverted when we unfocus from the inputs).

After weighing on the minor change in #9333 which introduced this issue, and the changes in #9630 which seems to be more substantial, I opted to take a step back and only revert the change made in #9333, without any changes in #9630.

With that said, I have also tested the 3 issues that were outlined in #9210 in the current latest codebase, and it seems like all 3 of them have been fixed (at least from what I tested), here's a quick clip:

https://user-images.githubusercontent.com/42867097/141777476-3cac6276-6db5-49f4-8604-b6d44f122448.mp4

## Changes made

- basically just reverts the addition of `|| typeof item === 'object'` in line 283
- removed `useRelationsStore()` which is actually unused, since it's currently using the composable m2m `useRelation` instead
- minor lint warning fix with `error: any`.

### CODE REVIEW

**Review by joselcvarela — APPROVED** [review]
Seems great to me.
Everything worked as expected. No blur occurred and everything was saved properly.

**Comment by joselcvarela** [comment]
I found a minor issue.
If you partial erase the translation (using backspace) and then put the same name as it was before, the field will blur.
It's not critical but would be cool to not have that behaviour.

**Review by joselcvarela — COMMENTED** [review]
I found a minor issue.
If you partial erase the translation (using backspace) and then put the same name as it was before, the field will blur.
It's not critical but would be cool to not have that behaviour.

**Comment by christianrr** [comment]
The fix in #9803 works as expected - many thanks 👍. As reported by @joselcvarela, the translation interface fields still blur, whenever Backspace is pressed.

It would be great if this issue could be reopened and this be fixed as well, since it's still a bit of an annoying experience for editors when inputting data.

**Comment by rijkvanzanten** [comment]
Lets open a new issue for that @christianrr @joselcvarela 👍🏻
